UnitySDK/Assets/ML-Agents/Scripts/InferenceBrain/MetaDataLoader.cs

```diff
@@ -0,0 +1,485 @@
+        /// <returns></returns>
+        private int GetIntScalar(string name)
+        {
+            var outputs = new Tensor[]
+            {
+                new Tensor()
+                {
+                    Name = name,
+                    ValueType = Tensor.TensorType.Integer,
+                    Shape = new long[] { },
+                    Data = new long[1]
+                },
+            };
+            try
+            {
+                _engine.ExecuteGraph(new Tensor[0], outputs);
+            }
+            catch (Exception e)
+            {
+                Debug.Log("Node not in graph: " + name + ". The following error occured : \n" + e);
+                return -1;
+            }
+
+            return (outputs[0].Data as int[])[0];
+        }
+
+        /// <summary>
+        /// Retrieves an IEnumerable of string corresponding to the failed compatibility checks
+        /// between the InferenceEngine and the BrainParameters. 
+        /// </summary>
+        public IEnumerable<string> GetChecks()
+        {
+            return _failedModelChecks;
+        }
+
+        /// <summary>
+        /// Generates the list of failed checks that failed when comparing the data from the Model
+        /// and from the BrainParameters
+        /// </summary>
+        private void GenerateChecks()
+        {
+            _failedModelChecks.Clear();
+            if (_engine == null)
+            {
+                _failedModelChecks.Add(
+                    "There is no model for this Brain, cannot run inference. " +
+                    "(But can still train)");
+                return;
+            }
+
+            var modelApiVersion = GetIntScalar(TensorNames.VersionNumber);
+            var memorySize = GetIntScalar(TensorNames.MemorySize);
+            var isContinuous = GetIntScalar(TensorNames.IsContinuousControl);
+            var actionSize = GetIntScalar(TensorNames.ActionOutputShape);
+            if (modelApiVersion == -1)
+            {
+                _failedModelChecks.Add(
+                    "Model was not trained using the right version of ML-Agents. Cannot use this " +
+                    "model.");
+                return;
+            }
+            if (modelApiVersion != ApiVersion)
+            {
+                _failedModelChecks.Add(
+                    $"Version of the trainer the model was trained with ({modelApiVersion}) " +
+                    $"is not compatible with the Brain's version ({ApiVersion}).");
+                return;
+            }
+
+            CheckIntScalarPresenceHelper(memorySize, isContinuous, actionSize);
+            CheckInputTensorPresence(memorySize, isContinuous);
+            CheckOutputTensorPresence(memorySize);
+            CheckInputTensorShape();
+            CheckOutputTensorShape(isContinuous, actionSize);
+        }
+
+        private void CheckIntScalarPresenceHelper(int memorySize, int isContinuous, int actionSize)
+        {
+            if (memorySize == -1)
```

Comment:
If the brain is not recurrent that's ok, no?

Reply:
No, It will have a value of 0 then.

Reply:
Gotcha, sounds good.